SD-K9/src/app/providers/map-coordinator.service.ts

```diff
@@ -48,14 +48,16 @@ export class MapCoordinator {
     }
 
     private _parseLocation(location: string): FloorPlanIdentifier | string {
-        if (location && location.substr(1,1) === '-') {
-            switch (location.substr(0,1)) {
+        if (location && location.indexOf("-") !== -1) {
+            let sliceIndex: number = location.indexOf("-");
+            let floorIndex: number = sliceIndex + 1;
+            switch (location.slice(0,sliceIndex)) {
                 case 'H': {
-                    let floorPlanIdentifier: FloorPlanIdentifier = {id: 1, building: 'hall', floor: +location.substr(2,1)};
+                    let floorPlanIdentifier: FloorPlanIdentifier = {id: 1, building: 'hall', floor: +location.substr(floorIndex,1)};
                     return floorPlanIdentifier;
                 }
-                case 'L': {
-                    let floorPlanIdentifier: FloorPlanIdentifier = {id: 1, building: 'loyola', floor: +location.substr(2,1)};
+                case 'CC': {
```

Comment:
What does CC stand for?

Reply:
It's the acronym the Loyola campus uses for its central building

Reply:
Well that's definitely interesting